app/frontend/src/features/profile/HostingPreferenceForm.tsx

```diff
@@ -1,10 +1,232 @@
-import { makeStyles } from "@material-ui/core";
-import React from "react";
+import {
+  Checkbox,
+  FormControl,
+  FormControlLabel,
+  makeStyles,
+  Typography,
+} from "@material-ui/core";
+import { Autocomplete } from "@material-ui/lab";
+import { unwrapResult } from "@reduxjs/toolkit";
+import React, { useState } from "react";
+import { Controller, useForm, UseFormMethods } from "react-hook-form";
+import { updateHostingPreference } from "./index";
+import Alert from "../../components/Alert";
+import Button from "../../components/Button";
+import { smokingLocationLabels } from "./constants";
+import ProfileTextInput from "./ProfileTextInput";
+import { ProfileFormData } from "../../service/user";
+import { useAppDispatch, useTypedSelector } from "../../store";
+import { SmokingLocation } from "../../pb/api_pb";
 
-const useStyles = makeStyles({
-  root: {},
-});
+interface HostingPreferenceCheckboxProps {
```

Comment:
I think it is best extend material-ui's CheckboxProps. What do you think about that?

Reply:
I think not if we keep it here for now, as the component here is opinionated and specific to this form only, so it wouldn't make sense to pollute the component props API surface. But otherwise I'd agree with that if this is meant to be a more generic component.